src/components/Spinner/Spinner.styles.ts

```diff
@@ -0,0 +1,43 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { addAlpha, getTheme } from '../../util/colors';
+import { mergeStyleSets } from '@uifabric/styling';
+import { memoizeFunction } from '@uifabric/utilities';
+import { SpinnerColor, SpinnerSize } from './types';
+
+export interface SpinnerClassNames {
+  root: string;
+  label: string;
+}
+
+const labelMargins = {
+  [SpinnerSize.XSMALL]: '-1px 0 0 6px',
+  [SpinnerSize.SMALL]: '-1px 0 0 8px',
+  [SpinnerSize.MEDIUM]: '-2px 0 0 8px',
+  [SpinnerSize.LARGE]: '4px 0 0',
+};
+
+export const getClassNames = memoizeFunction((size: SpinnerSize, isCentered: boolean): SpinnerClassNames => {
+  return mergeStyleSets({
+    root: {
+      display: 'flex',
+      alignItems: 'center',
+      flexDirection: size === SpinnerSize.LARGE ? 'column' : 'row',
+      justifyContent: isCentered ? 'center' : undefined,
+    },
+    label: {
+      margin: labelMargins[size],
+    },
+  });
+});
+
+export const getFabricSpinnerStyles = memoizeFunction((color: SpinnerColor) => {
+  const theme = getTheme();
+  const circleColor = color === SpinnerColor.DARK ? theme.palette.white : theme.semanticColors.link;
```

Comment:
hey @chrismohr, take a look at this. it pulls the current theme directly from Fabric instead of from our `<Partial>` theme colors. what do you think about this approach?

Reply:
@swese44 I like it! It'll be nice to have all of the colors available to us from the same source of truth like this, instead of just the subset that we're overriding. 

And I also like how you're importing it from util/colors, instead of directly  because I think we'll probably want to extend the fabric theme to have some of our own semantic color names.  In which case we'll be able to wrap it pretty easily.

Reply:
@swese44 I think I'd advocate for using a differenent color name here though.

My votes would be to use `themeDark` or create our own semantic slot called something like `spinner` and set it to `themeDark`.

Reason being, is that I think we should try to minimize potential surprises.

If I were a developer using this library, I would expect that if I changed `link` color to green, only the links would change to green.

I would be surprised if the spinner also changed color.

If instead I wanted to turn everything river colored, into green, I would override `themeDark` from the palette.

Reply:
It just occurred to me, that we should be consuming font sizes and lineHeights in the same way.  You may have pointed this out before, and I missed it, but we've defined them in both https://github.com/Microsoft/YamUI/blob/master/src/util/styles/fonts.ts#L8-L26 and https://github.com/Microsoft/YamUI/blob/master/src/util/fabric/theme.ts#L14-L40